packages/client/ui/react-ui/src/components/embed/v3/crypto/utils/handleSendTransaction.ts

```diff
@@ -0,0 +1,31 @@
+import type { Wallet } from "@dynamic-labs/sdk-react-core";
+
+import type { BlockchainIncludingTestnet } from "@crossmint/common-sdk-base";
+
+import { handleEvmTransaction } from "./handleEvmTransaction";
+
+import { isSolanaWallet } from "@dynamic-labs/solana";
+import { handleSolanaTransaction } from "./handleSolanaTransaction";
+import { isEthereumWallet } from "@dynamic-labs/ethereum";
+import type { EmbeddedCheckoutV3IFrameEmitter } from "@crossmint/client-sdk-base";
+
+export async function handleSendTransaction(
+    primaryWallet: Wallet,
+    chain: BlockchainIncludingTestnet,
+    serializedTransaction: string,
+    iframeClient: EmbeddedCheckoutV3IFrameEmitter
+) {
+    const commonParams = {
+        chain,
+        serializedTransaction,
+        iframeClient,
+    };
+    if (isSolanaWallet(primaryWallet)) {
+        return await handleSolanaTransaction({
+            ...commonParams,
+            primaryWallet,
+        });
```

Comment:
They reuse the entire object right? `commonParams` could include primaryWallet as well I think

Reply:
its a ts thing, cuz of the `isSolanaWallet` and `isEthereumWallet` so it has to be after commonParams